### ISSUE TITLE
[bitnami/kube-prometheus] Unable to scrap the kubelet process where the prometheus-kubelet pod is deployed

### ISSUE THREAD

**Issue by ausias-armesto** [bitnami/charts#13548]
### Name and Version

bitnami/prometheus 8.1.12

### What steps will reproduce the bug?

1. With a Kubespray distribution on Hetzner
2. I install the helm chart kube-prometheus
3. Using the below values file
4. I cannot scrap the metrics from the kubelet process where the pod `prometheus-prometheus-kube-prometheus-prometheus-0` is deployed.
If I cordon the node an force a redeployment of the statefulset, to be deployed in other node, then the other node is failing to scrap the metrics, when before it was getting them correctly.

### Are you using any custom parameters or values?

````
        operator:
          logLevel: debug
        clusterDomain: "cluster.{{ environment_name }}"
        prometheus:
          logLevel: debug
          ingress:
            enabled: true
            hostname: "prometheus.{{ letsencrypt_domain }}"
            path: /
            ingressClassName: nginx
        alertmanager:
          logLevel: debug
        kubeControllerManager:
          serviceMonitor:
            https: true
            insecureSkipVerify: true
          service:
            ports:
              http: 10257
            targetPorts:
              http: 10257
        kubeScheduler:
          serviceMonitor:
            https: true
            insecureSkipVerify: true
          service:
            ports:
              http: 10259
            targetPorts:
              http: 10259
        kubeProxy:
          serviceMonitor:
            https: false
````

### What is the expected behavior?

I expect to collect metrics from kubelet from all nodes, no matter where it has been deployed the pod `prometheus-prometheus-kube-prometheus-prometheus-0`

### What do you see instead?

The message error is saying that there is a timeout

````
Get "https://10.0.1.3:10250/metrics": context deadline exceeded
````

### Additional information

I have been able to deploy a simple pod on the same server as it is deployed the `prometheus-prometheus-kube-prometheus-prometheus-0` pod and I have been able to gather metrics with the following command:
````
cat <<EOF > ubuntu-tool.yaml
apiVersion: v1
kind: Pod
metadata:
  name: ubuntu-tool
spec:
  containers:
    - name: ubuntu
      image: ubuntu:latest
      command:
        - "sleep"
        - "604800"
      imagePullPolicy: IfNotPresent
      volumeMounts:
        - name: certificates
          mountPath: /etc/kubernetes
  volumes:
    - name: certificates
      hostPath:
        path: /etc/kubernetes
        type: DirectoryOrCreate      
  restartPolicy: Always      
EOF
````

````
apt-get update && apt install -y curl
curl https://10.0.0.3:10250/metrics -v -k --cert /etc/kubernetes/ssl/apiserver-kubelet-client.crt --key /etc/kubernetes/ssl/apiserver-kubelet-client.key --cacert /etc/kubernetes/ssl/ca.crt
````

- I have enable the logs but don't see any message out of normal
- I have seen the kubelet logs of the node, and also don't see neither any anomaly there.
- The network plugin used is calico, but in the logs I don't see nothing neither
- I have checked the firewall as well, but is not blocking anything. Although should be that, as the request should be handled directly by the node without the need to do an external request.

**Comment by javsalgar** [comment]
Hi,

Could you try adding an ephemeral container? 

https://kubernetes.io/docs/tasks/debug/debug-application/debug-running-pod/#ephemeral-container

**Comment by ausias-armesto** [comment]
Hi @javsalgar ,

I found the reason of the failure. The root cause was that in the Kubespray distribution I had set the parameter `kubelet_secure_addresses` too restrictive so few ip ranges could access the kubelet as it is recommended by [Hardening guide](https://github.com/kubernetes-sigs/kubespray/blob/master/docs/hardening.md).
Once I changed the parameter to allow the kubernetes pod ip range to access the kubelet, then the prometheus pod worked smoothly.

I will close this issue, and would like to thank you for your help.
